scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,189 @@
+from typing import Optional
+
+# Workaround to use the preview SDK
+from azure.search.documents.indexes._generated.models import (
+    IndexProjectionMode,
+    SearchIndexerIndexProjections,
+    SearchIndexerIndexProjectionSelector,
+    SearchIndexerIndexProjectionsParameters,
+    SearchIndexerSkillset,
+)
+from azure.search.documents.indexes.models import (
+    AzureOpenAIEmbeddingSkill,
+    AzureOpenAIParameters,
+    AzureOpenAIVectorizer,
+    FieldMapping,
+    InputFieldMappingEntry,
+    OutputFieldMappingEntry,
+    SearchIndexer,
+    SearchIndexerDataContainer,
+    SearchIndexerDataSourceConnection,
+    SplitSkill,
+)
+
+from .blobmanager import BlobManager
+from .embeddings import AzureOpenAIEmbeddingService
+from .listfilestrategy import ListFileStrategy
+from .searchmanager import SearchManager
+from .strategy import DocumentAction, SearchInfo, Strategy
+
+
+class IntegratedVectorizerStrategy(Strategy):
+    """
+    Strategy for ingesting and vectorizing documents into a search service from files stored storage account
+    """
+
+    def __init__(
+        self,
+        list_file_strategy: ListFileStrategy,
+        blob_manager: BlobManager,
+        embeddings: Optional[AzureOpenAIEmbeddingService],
+        subscriptionId: str,
+        searchServiceUserAssginedId: str,
+        document_action: DocumentAction = DocumentAction.Add,
+        search_analyzer_name: Optional[str] = None,
+        use_acls: bool = False,
+        category: Optional[str] = None,
+    ):
+        self.list_file_strategy = list_file_strategy
+        self.blob_manager = blob_manager
+        self.document_action = document_action
+        self.embeddings = embeddings
+        self.subscriptionId = subscriptionId
+        self.search_user_assigned_identity = searchServiceUserAssginedId
+        self.search_analyzer_name = search_analyzer_name
+        self.use_acls = use_acls
+        self.category = category
+
+    async def createEmbeddingSkill(self, index_name: str):
```

Comment:
```suggestion
    async def create_embedding_skill(self, index_name: str):
```

snake case for function names